angrmanagement/ui/dialogs/preferences.py

```diff
@@ -177,9 +180,17 @@ def _init_widgets(self):
         log_format_lbl = QLabel("Log datetime Format String:")
         log_format_lbl.setSizePolicy(QSizePolicy(QSizePolicy.Fixed, QSizePolicy.Fixed))
         log_format_layout.addWidget(log_format_lbl)
-        self.log_format_entry = QLineEdit(self)
-        self.log_format_entry.setClearButtonEnabled(True)
-        self.log_format_entry.setText(Conf.log_timestamp_format)
+
+        self.log_format_entry = QComboBox(self)
+        fmt: str = Conf.log_timestamp_format
+        ts = datetime.now()
+        # pylint: disable=use-sequence-for-iteration
+        self._fmt_map = bidict({ ts.strftime(i): i for i in {fmt, "%X", "%c"} })
```

Comment:
I still think using a set to implicitly de-duplicate values is not a great pattern

Reply:
It's what a set is for in this case; you care about the items being unique, not the order or anything; that's the set data structure use-case.